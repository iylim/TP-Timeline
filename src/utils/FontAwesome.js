import { library, config } from '@fortawesome/fontawesome-svg-core';
import { faStarOfLife } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

config.autoAddCss = false
library.add( faStarOfLife );

export { FontAwesomeIcon };