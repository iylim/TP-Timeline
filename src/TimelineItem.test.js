import React from 'react';
import { shallow } from 'enzyme';
import VerticalTimelineElement from './TimelineItem';
import { FontAwesomeIcon } from './utils/FontAwesome';

describe('VerticalTimelineElement', () => {
  it('should render correctly without props', () => {
    const component = shallow(<VerticalTimelineElement/>);
    expect(component).toMatchSnapshot();
  })

  it('should render date', () => {
    const date = 'Oct 10';
    const component = shallow(<VerticalTimelineElement date={date} />);
    expect(component).toMatchSnapshot();
  })

  it('should render icon with correct styles',() => {
    const icon = <FontAwesomeIcon icon={['fab', 'node-js']} />;
    const iconStyle = { background: '#FA7268', color: '#fff' };
    const component = shallow(<VerticalTimelineElement icon={icon} iconStyle={iconStyle} />);
    expect(component).toMatchSnapshot();
  })
})