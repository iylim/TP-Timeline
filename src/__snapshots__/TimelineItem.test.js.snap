// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerticalTimelineElement should render correctly in Debug 1`] = `
<div
  className="vertical-timeline-element vertical-timeline-element--no-children"
  id=""
  style={null}
>
  <VisibilitySensor
    active={true}
    containment={null}
    delayedCall={false}
    intervalCheck={true}
    intervalDelay={100}
    minTopValue={0}
    offset={
      Object {
        "bottom": 80,
      }
    }
    onChange={[Function]}
    partialVisibility={true}
    resizeCheck={false}
    resizeDelay={250}
    resizeThrottle={-1}
    scrollCheck={false}
    scrollDelay={250}
    scrollThrottle={-1}
  >
    <div>
      <span
        className="vertical-timeline-element-icon is-hidden"
        onClick={null}
        style={null}
      />
      <div
        className="vertical-timeline-element-content is-hidden"
      >
        <span
          className="vertical-timeline-element-date"
        />
      </div>
    </div>
  </VisibilitySensor>
</div>
`;

exports[`VerticalTimelineElement should render correctly without props 1`] = `
<div
  className="vertical-timeline-element vertical-timeline-element--no-children"
  id=""
  style={null}
>
  <VisibilitySensor
    active={true}
    containment={null}
    delayedCall={false}
    intervalCheck={true}
    intervalDelay={100}
    minTopValue={0}
    offset={
      Object {
        "bottom": 80,
      }
    }
    onChange={[Function]}
    partialVisibility={true}
    resizeCheck={false}
    resizeDelay={250}
    resizeThrottle={-1}
    scrollCheck={false}
    scrollDelay={250}
    scrollThrottle={-1}
  >
    <div>
      <span
        className="vertical-timeline-element-icon is-hidden"
        onClick={null}
        style={null}
      />
      <div
        className="vertical-timeline-element-content is-hidden"
      >
        <span
          className="vertical-timeline-element-date"
        />
      </div>
    </div>
  </VisibilitySensor>
</div>
`;

exports[`VerticalTimelineElement should render date 1`] = `
<div
  className="vertical-timeline-element vertical-timeline-element--no-children"
  id=""
  style={null}
>
  <VisibilitySensor
    active={true}
    containment={null}
    delayedCall={false}
    intervalCheck={true}
    intervalDelay={100}
    minTopValue={0}
    offset={
      Object {
        "bottom": 80,
      }
    }
    onChange={[Function]}
    partialVisibility={true}
    resizeCheck={false}
    resizeDelay={250}
    resizeThrottle={-1}
    scrollCheck={false}
    scrollDelay={250}
    scrollThrottle={-1}
  >
    <div>
      <span
        className="vertical-timeline-element-icon is-hidden"
        onClick={null}
        style={null}
      />
      <div
        className="vertical-timeline-element-content is-hidden"
      >
        <span
          className="vertical-timeline-element-date"
        >
          Oct 10
        </span>
      </div>
    </div>
  </VisibilitySensor>
</div>
`;

exports[`VerticalTimelineElement should render icon with correct styles 1`] = `
<div
  className="vertical-timeline-element vertical-timeline-element--no-children"
  id=""
  style={null}
>
  <VisibilitySensor
    active={true}
    containment={null}
    delayedCall={false}
    intervalCheck={true}
    intervalDelay={100}
    minTopValue={0}
    offset={
      Object {
        "bottom": 80,
      }
    }
    onChange={[Function]}
    partialVisibility={true}
    resizeCheck={false}
    resizeDelay={250}
    resizeThrottle={-1}
    scrollCheck={false}
    scrollDelay={250}
    scrollThrottle={-1}
  >
    <div>
      <span
        className="vertical-timeline-element-icon is-hidden"
        onClick={null}
        style={null}
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Array [
              "fab",
              "node-js",
            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
      </span>
      <div
        className="vertical-timeline-element-content is-hidden"
      >
        <span
          className="vertical-timeline-element-date"
        />
      </div>
    </div>
  </VisibilitySensor>
</div>
`;
